### PR TITLE
Integrates sponsors API to sponsors route

### LINKED ISSUE #579: Implement sponsors API to sponsors-step form
Parent issue #462

### PR DESCRIPTION
**Pull request by geekyd**
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:

- Integrates sponsors API to sponsors route
- Modifies link-input for valid urls

Fixes #579

### CODE REVIEW

**Comment by geekyd** [comment]
This will be merged after https://github.com/fossasia/open-event-orga-server/issues/4201 & https://github.com/fossasia/open-event-orga-server/issues/4203 are resolved on the server.

**Comment by geekyd** [comment]
@niranjan94 This is working now, as expected however there is one thing I needed to confirm before this merges, will we be deleting all the sponsors if the user turns the sponsors off?

**Comment by niranjan94** [comment]
@geekyd yes. We will be.

**Comment by geekyd** [comment]
@niranjan94 Please review. Everything seems to work now. Thanks

**Comment by niranjan94** [comment]
Deployment link: https://pr-580-open-event-frontend-fossasia.surge.sh

**Comment by niranjan94** [comment]
Deployment link: https://pr-580-open-event-frontend-fossasia.surge.sh

File: app/components/forms/wizard/sponsors-step.js [hunk 1]
```diff
@@ -43,8 +44,15 @@ export default Component.extend(FormMixin, {
...
     publish() {
       this.onValid(() => {
+        this.set('data.event.state', 'published');
         this.get('save')('publish');
       });
     }
+  },
+
+  didInsertElement() {
+    if (this.get('data.event.sponsors') && !this.get('data.event.sponsors').toArray().length) {
```
Comment (niranjan94) [inline app/components/forms/wizard/sponsors-step.js]: data.event.sponsors.length will give you length. No need to convert to array.

**Comment by niranjan94** [comment]
Deployment link: https://pr-580-open-event-frontend-fossasia.surge.sh

File: app/components/forms/wizard/sponsors-step.js (hunk 1 above)
Comment (geekyd) [inline app/components/forms/wizard/sponsors-step.js]: Done!

**Pull request merged**